File: app/android/app/src/main/java/com/jun/weather/repository/web/api/ShortWeatherInfoService.java
package com.jun.weather.repository.web.api;

import com.jun.weather.repository.web.api.entity.ApiResponse;
import com.jun.weather.repository.web.api.entity.ForecastItem;
import com.jun.weather.repository.web.api.entity.MidLandItem;
import com.jun.weather.repository.web.api.entity.MidTempItem;
import com.jun.weather.repository.web.api.entity.ObserveItem;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface ShortWeatherInfoService {
    @GET("getUltraSrtNcst")
    Call<ApiResponse<ObserveItem>> getUltraSrtNcst(@QueryMap Map<String, Object> query);

    @GET("getVilageFcst")
    Call<ApiResponse<ForecastItem>> getVilageFcst(@QueryMap Map<String, Object> query);

    @GET("getUltraSrtFcst")
    Call<ApiResponse<ForecastItem>> getUltraSrtFcst(@QueryMap Map<String, Object> query);
}

File: app/android/app/src/main/java/com/jun/weather/repository/web/api/ShortWeatherInfoService.kt
package com.jun.weather.repository.web.api

import com.jun.weather.repository.web.entity.ApiResponse
import com.jun.weather.repository.web.entity.ForecastItem
import com.jun.weather.repository.web.entity.ObserveItem
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface ShortWeatherInfoService {
    @GET("getUltraSrtNcst")
    fun getUltraSrtNcst(@QueryMap query: Map<String, String>): Call<ApiResponse<ObserveItem>>

    @GET("getVilageFcst")
    fun getVilageFcst(@QueryMap query: Map<String, String>): Call<ApiResponse<ForecastItem>>

    @GET("getUltraSrtFcst")
    fun getUltraSrtFcst(@QueryMap query: Map<String, String>): Call<ApiResponse<ForecastItem>>
}